.github/workflows/gradle.yml

```diff
@@ -0,0 +1,53 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: '*'
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out this repo
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        repository: ruderphilipp/jmc.git
+        ref: main
+    - name: Check out the triggering repository
+      run: |
+         declare -A SUBMODULES=( \
+            [jameica]=jameica \
+            [datasource]=lib/datasource \
+            [util]=lib/util \
+            [example]=plugins/example \
+            [hibiscus]=plugins/hibiscus \
+            )
+
+         set -x
+         REPONAME=${GITHUB_REPOSITORY#*/}
+         if [ ${REPONAME} = "jmc" ]; then
+            echo "Github Actions run started from this repository"
+         else
+            git -C ${SUBMODULES[$REPONAME]} remote add action_repo git@github.com:${GITHUB_REPOSITORY}
+            git -C ${SUBMODULES[$REPONAME]} fetch action_repo ${GITHUB_SHA}
+            git -C ${SUBMODULES[$REPONAME]} checkout ${GITHUB_SHA}
+         fi
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+    - name: Build with Gradle
+      run: gradle build
+    - name: upload package
+      uses: actions/upload-artifact@master
+      with:
+        name: pkg
+        path: 'jameica/build/gradle/distributions/*.zip'
```

Comment:
Hier wird die Datei vom Projekt `jameica` hochgeladen --> Das hier ist aber das Projekt `hibiscus`

Aus meiner Sicht ist die Frage, ob es sinnvoller ist, `plugins/hibiscus/build/target/*.jar` als Minimalpaket hochzuladen oder ob man hier eine FatJar mit allen Abhängigkeiten als Paket erstellen sollte.

Reply:
Das ist eine gute Frage, für die einzelnen Build-Artefakte sind die Target-Jars wahrscheinlich sinnvoller.

Reply:
Ist in d8ee11de5aefcb628e344194c3ea17e89759ac65 geändert: es werden nur die target-jars vom auslösenden Repository hochgeladen